roles/idrac_bios/molecule/clear_pending_attributes/prepare.yml

```diff
@@ -0,0 +1,58 @@
+---
+- name: Converge file to update attributes with apply time as Immediate
+  hosts: all
+  gather_facts: false
+  vars:
+    idrac_ip: "{{ lookup('env', 'IDRAC_IP') }}"
+    idrac_port: "{{ lookup('env', 'IDRAC_PORT') }}"
+  tasks:
+    - name: Fetch Jobs Data
+      ansible.builtin.include_tasks:
+        file: ../__get_data.yml
+      vars:
+        url: https://{{ idrac_ip }}:{{ idrac_port }}/redfish/v1/Managers/iDRAC.Embedded.1/Jobs?$expand=*($levels=1)
+
+    - name: Fetch Bios Jobs Data
+      when: idrac_bios_uri_data.json.Members | length > 0
+      ansible.builtin.set_fact:
+        idrac_bios_jobs_items: "{{ idrac_bios_uri_data.json.Members | json_query(query) }}"
+      vars:
+        query: "[?JobType=='BIOSConfiguration' && JobState=='Scheduled' || JobState=='Scheduling' ]"
+      no_log: true
+
+    - name: Block for creating a bios job as a pre-requisite
+      when: idrac_bios_jobs_items | length == 0
+      block:
+        - name: Fetch IDRAC Data
+          ansible.builtin.include_tasks:
+            file: ../__get_data.yml
+          vars:
+            url: https://{{ idrac_ip }}:{{ idrac_port }}/redfish/v1/Systems/System.Embedded.1/Bios
+
+        - name: Fetch the existing boot mode
+          ansible.builtin.set_fact:
+            current_boot_mode: "{{ idrac_bios_uri_data.json.Attributes.BootMode }}"
+
+        - name: Set the boot mode
+          ansible.builtin.set_fact:
+            set_boot_mode: "{{ 'Bios' if (current_boot_mode == 'Uefi') else 'Uefi' }}"
```

Comment:
can these vars not be defined as idrac_bios_set_boot_mode, idrac_bios_current_boot_mode.. similar to the others?

Reply:
Not possible as that is giving sonar issues if we rename it.